src/utils/encoding.ts

```diff
@@ -0,0 +1,64 @@
+import * as jschardet from "jschardet"
+import * as iconv from "iconv-lite"
+import { isBinaryFile } from "isbinaryfile"
+import fs from "fs/promises"
+import path from "path"
+
+/**
+ * Detect the encoding of a file buffer
+ * @param fileBuffer The file buffer
+ * @param fileExtension Optional file extension
+ * @returns The detected encoding
+ */
+export async function detectEncoding(fileBuffer: Buffer, fileExtension?: string): Promise<string> {
+	// 1. First check if it's a binary file
+	if (fileExtension) {
+		const isBinary = await isBinaryFile(fileBuffer).catch(() => false)
+		if (isBinary) {
+			throw new Error(`Cannot read text for file type: ${fileExtension}`)
+		}
+	}
+
+	// 2. Perform encoding detection
+	const detected = jschardet.detect(fileBuffer)
+	let encoding: string
+	let originalEncoding: string | undefined
+	
+	if (typeof detected === "string") {
+		encoding = detected
+		originalEncoding = detected
+	} else if (detected && detected.encoding) {
+		originalEncoding = detected.encoding
+		// Check confidence level, use default encoding if too low
+		if (detected.confidence < 0.7) {
```

Comment:
Could we add a comment explaining why 0.7 was chosen as the threshold? Based on your explanation in the PR comments, perhaps something like:
```suggestion
		// Check confidence level, use default encoding if too low
		// 0.7 is a conservative threshold that works well when UTF-8 is the dominant encoding
		// and we prefer to fall back rather than risk mis-decoding
		if (detected.confidence < 0.7) {
```